internal/app/secretless/providers/conjur/provider.go

```diff
@@ -191,69 +196,24 @@ func loadAuthenticator(authnURL string, version string, tokenFile string, config
 		}
 	}
 
-	// Load configuration from the environment
-	// TODO get pod namespace / name using Kubernetes API
-	// instead of specifying in deployment manifest
-	podNamespace := os.Getenv("MY_POD_NAMESPACE")
-	podName := os.Getenv("MY_POD_NAME")
-	account := os.Getenv("CONJUR_ACCOUNT")
-	authnLogin := os.Getenv("CONJUR_AUTHN_LOGIN")
-
-	// Load CA cert
-	if conjurCACert, err = readSSLCert(); err != nil {
+	config, err := authnConfig.NewFromEnv(&clientCertPath, &tokenFilePath)
+	if err != nil {
 		return nil, err
 	}
 
 	// Create new Authenticator
-	authenticator, err := authenticator.New(
-		authenticator.Config{
-			Account:        account,
-			ClientCertPath: clientCertPath,
-			ConjurVersion:  version,
-			PodName:        podName,
-			PodNamespace:   podNamespace,
-			SSLCertificate: conjurCACert,
-			TokenFilePath:  tokenFile,
-			URL:            authnURL,
-			Username:       authnLogin,
-		})
+	authenticator, err := authenticator.New(*config)
 	if err != nil {
 		return nil, err
 	}
 
-	// Send the login request to Conjur to retrieve the signed certificate
-	// Configure exponential backoff
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 2 * time.Second
-	expBackoff.RandomizationFactor = 0.5
-	expBackoff.Multiplier = 2
-	expBackoff.MaxInterval = 15 * time.Second
-	expBackoff.MaxElapsedTime = 2 * time.Minute
-
-	// Try login with exponential backoff on failure
-	err = backoff.Retry(
-		func() error {
-			if err = authenticator.Login(); err != nil {
-				return err
-			}
-
-			return nil
-		},
-		expBackoff)
-
-	// If unable to login, return error
-	if err != nil {
-		return nil, fmt.Errorf("Error: Conjur provider unable to log in to Conjur: %s", err.Error())
-	}
-
 	return authenticator, nil
 }
 
 // refreshAccessToken uses the Conjur Kubernetes authenticator
 // to authenticate with Conjur and retrieve a new time-limited
 // access token in a loop
-func (p *Provider) refreshAccessToken() error {
-
+func (p *Provider) refreshAccessToken(oneShot bool) error {
```

Comment:
I'd rather call this something clearer (like `returnAfterOneRefresh` maybe?) but not a blocker

Reply:
fixed by @doodlesbykumbi PR